storage/database/db_manager.go

```diff
@@ -476,12 +476,76 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 		defer dbm.lockInMigration.RUnlock()
 
 		if dbm.inMigration {
-			return dbm.GetStateTrieMigrationDB().NewBatch()
+			newDBBatch := dbm.getDatabase(StateTrieMigrationDB).NewBatch()
+			oldDBBatch := dbm.getDatabase(StateTrieDB).NewBatch()
+			return NewStateTrieDBBatch([]Batch{oldDBBatch, newDBBatch})
```

Comment:
Is there any place we need to write data to both of oldDB and migrationDB?

Reply:
For recovering old database, this PR introduce this batch in migration.
This is used for `trieDB.Commit` / `trieDB.Cap`.